### PR TITLE
devguide: move into userguide as last chapter - v3

### PR DESCRIPTION
**Pull request by jufajardini**
This still fails with make distcheck.
I tried adding `tools/generate-images.sh` directly as a `EXTRA_DIST`, but it still didn't work, so this version only lists the directory for the script. I noticed that when the failure occurs the build command seems to be called from a deeper subdir, somehow, and wonder if this could be affecting, even though the files should have been copied for the dist package.

I believe we didn't see this failure happening in master because the command is run for the userguide only, and therefore the script wasn't being invoked before.

Describe changes:
- added a commit to fix the build of the devguide in the dist package.

Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/4909

Previous PR: https://github.com/OISF/suricata/pull/6917

Preview at:
https://suri-rtd-test.readthedocs.io/en/latest/devguide/

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -221,6 +221,9 @@ jobs:
                 sudo \
                 which \
                 zlib-devel
+          # Packages needed for some diagrams generation.
+          dnf -y install epel-release
+          dnf -y install mscgen
           # These packages required to build the PDF.
           yum -y install \
```
Comment (inashivb) [inline .github/workflows/builds.yml:228]: Q: why do we use `yum` for all the packages?

**Review by inashivb — COMMENTED** [review]
Q: Do we need the static files again in devguide? Do we change something there or is it required by RTD?

**Comment by jufajardini** [comment]
> Q: Do we need the static files again in devguide? Do we change something there or is it required by RTD?

I think that's something that the `conf.py` script is doing. Currently I've let the devguide in such a way that it could be built and generated independently, if wanted, and so the conf.py, among other files, stayed. If this is judged unnecessary, then I think we could/would also remove those files....

File: .github/workflows/builds.yml (hunk 1 above)
Comment (jasonish) [reply .github/workflows/builds.yml:228]: As of CentOS 8 (Alma, RHEL, etc) `yum` is just an alias to `dnf`.  When I wrote this, I probably still had the muscle memory from CentOS 7 days where there was only `yum`.

Comment (inashivb) [reply .github/workflows/builds.yml:228]: oooh didn't know it was an alias now. Thank you!

**Review by jasonish — CHANGES_REQUESTED** [review]
I think we should just focus on it being part of the main guide, and not worry about making the dev part being independently buildable - that just doubles the amount "infrastructure" we need to worry about, test, etc.

Has this been a specific request already?

**Comment by jufajardini** [comment]
> I think we should just focus on it being part of the main guide, and not worry about making the dev part being independently buildable - that just doubles the amount "infrastructure" we need to worry about, test, etc.
> 
> Has this been a specific request already?

Not that I have seen anywhere. Oki, I'll submit a new PR without those - but from my local tests I think this won't solve the make distcheck thing, yet :/

**Comment by jufajardini** [comment]
followed by: https://github.com/OISF/suricata/pull/6975

**Pull request closed without merging**